### ISSUE TITLE
Revise how Packages are Imported

### ISSUE THREAD

**Issue by curran** [datavis-tech/vizhub-issue-tracker#224]
* [x] Remove `<script>` tags from `index.html`.
 * [x] Emulate `package.json` format for dependencies (with versions specified).
 * [x] Auto-inject `<script>` tags based on dependencies.

Follow-on features:

 * Auto-suggest upgrading dependencies
 * Consider another viz as a dependency #219

**Comment by curran** [comment]
Some fascinating stuff in the Svelte REPL around this: https://github.com/sveltejs/svelte-repl/blob/9624a6e471c62a769b08a7a291146c48b06bc256/src/workers/bundler/index.js#L143

**Comment by curran** [comment]
One strong use case for this: in tutorials, educational material, **books**, where compatibility with industry standard is desired.

**Comment by curran** [comment]
Case to consider: exported thing should work.

So, in the exported thing, maybe create something like `index_original.html`, then replace `index.html` with the derived HTML that has the script tags injected.

**Comment by curran** [comment]
@stushurik more description and context:
 * "Remove <script> tags from index.html.", meaning, make it so that end users no longer need to manually add `<script>` tags to `index.html`. No tags would actually be removed, it would just introduce the option of following a new pattern where adding `<script>` tags that point to CDNs would not be required.
 * "Auto-inject <script> tags based on dependencies." -- users should not see auto-injected script tags, should they? No, they should not. The Runner component will take as input a generated HTML document source that is _different_ from what they see in the editor. It will be derived from the `index.html` that they author. Much in the same way `create-react-app` does it (I believe they inject into the HTML some `<script>` tags during the `npm run build` step).
 * "Add UI & new field for dependencies. -- What kind of ui would you like to add ? Something like codesandbox.io do?" Yes, something like that. We can consider addition of a UI to be a "Phase II" sort of thing. No need to worry about this UI now. I'd like to have the presence of a `package.json` be the entry point for enabling this new scheme. Once that is in place we can build a UI for editing it.

**Comment by curran** [comment]
This issue needs a lot of consideration.

It's very much related to
 * API to Read From and Write To Viz #488
 * Bl.ocks.org Compatibility #171
 * GitHub Integration Phase I #301

Another idea with this: we can hide `bundle.js` entirely, just like `create-react-app` does in development mode.

Deep consideration of the structure of the exported files is required.

After the changes that address this issue, it should be possible to:
 * keep all the existing vizzes running with no modification.
 * transform from the authored files in VizHub to the exported files
 * have the exported files hosted in a static file server and run correctly
 * have the exported files hosted in a Gist on bl.ocks.org and present correctly
 * transform from the exported files (potentially modified) _back into_ the authored files in VizHub

Suggestion: the new features around this issue are opt-in based on the presence or absence of `package.json`. In effect we can use the existence of `package.json` as a feature flag to enable this new set of features.

We need to be careful to maintain all of these constraints.

**Comment by curran** [comment]
Noting the status: the feature was implemented, but may have introduced some breakages, so I disabled it temporarily to unblock a deploy of the latest to production.

Considering adding Sentry to catch client-side errors. Created a new issue for it: Add Sentry to catch client-side errors #509.

Plan moving forward: enable the feature flag on staging and do rigorous manual QA against it. Potentially we can develop automated tests, which would make the `package.json` feature much more robust.

Another idea: Open Source the bundling and runtime mechanism, so that it can grow as its own standalone thing, and a community of developers can work to improve it over time, submit bug fixes, etc.
